### ISSUE TITLE
MP3 support

### ISSUE THREAD

**Issue by nomeata** [csabahruska/proteaaudio#12]
I’m considering using proteaaudio in one of my projects where I currently shell out to `sox`. One feature I’ll probably miss is support for MP3 in addition to WAV and OGG. Is that by any chance on your road map and I can just wait for that, or would I have to work around it anyways?

**Comment by csabahruska** [comment]
IMO I could add MP3 support. This lib looks promising: https://github.com/lieff/minimp3
I prefer single file solutions without external dependencies.

**Comment by nomeata** [comment]
That would be great!

**Comment by nomeata** [comment]
Although I just found some other problems (namely that libpulseaudio cannot really be linked statically), so I am not sure if I can actually use proteaaudio :-(. So no rush with the MP3 support from my side, I fear.

**Comment by csabahruska** [comment]
I've added MP3 support. Please check it and if it works for you then I'll make a new release on Hackage.
https://github.com/csabahruska/proteaaudio/commit/aa482067333b01361cc370571c900b76e2267fbb

**Comment by nomeata** [comment]
Works flawlessly!

In my application I need to handle different formats without knowing their filename, and will use logic like this
```
sampleFromMemory :: B.ByteString -> Float -> IO Sample
sampleFromMemory bs
  | B.take 4 bs == B.pack "RIFF" = sampleFromMemoryWav (B.toStrict bs)
  | B.take 4 bs == B.pack "OggS" = sampleFromMemoryOgg (B.toStrict bs)
  | B.take 3 bs == B.pack "ID3" || B.take 2 bs `elem`
      [B.pack "\xFF\xFB", B.pack "\xFF\xF3", B.pack "\xFF\xF2"]
  = sampleFromMemoryMp3 (B.toStrict bs)
  | otherwise = error "Could not detect audio format"
 where magic = B.take 4 bs
```
If you think that logic would actually be good to have in proteaaudio itself, I’ll of course prefer that :-)

Besides that I think I’m good for now, thanks!

**Comment by csabahruska** [comment]
Released [proteaaudio-0.10.1](https://hackage.haskell.org/package/proteaaudio-0.10.1) with mp3 support and `sampleFromMemory`.
